pymatgen/analysis/phase_diagram.py

```diff
@@ -319,10 +353,9 @@ def from_entries(cls, entries, elements=None):
     @classmethod
     def _kwargs_from_entries(cls, entries, elements):
         if elements is None:
-            elements = set()
-            for entry in entries:
-                elements.update(entry.composition.elements)
-            elements = sorted(list(elements))
+            elements = sorted(list(set().union(
```

Comment:
I believe this can just be `sorted({els for e in entries for els in e.composition.elements})`